packages/manager-react-components/src/hooks/datagrid/useIcebergV6.tsx

```diff
@@ -24,73 +19,48 @@ export function useResourcesIcebergV6<T = unknown>({
   queryKey,
   defaultSorting = undefined,
 }: IcebergFetchParamsV6 & IcebergV6Hook) {
-  const [totalCount, setTotalCount] = useState(0);
-  const [flattenData, setFlattenData] = useState<T[]>([]);
-  const [queryParams, setQueryParams] = useState<QueryParams>({
-    pageIndex: 1,
-    sorting: { desc: false, id: '' },
-  });
   const [sorting, setSorting] = useState<ColumnSort>(defaultSorting);
 
   const {
-    data,
+    data: dataSelected,
     fetchNextPage,
     isError,
     isLoading,
     error,
     status,
   } = useInfiniteQuery({
-    initialPageParam: null,
+    initialPageParam: 1,
     queryKey: [...queryKey, sorting],
-    queryFn: ({ pageParam }) =>
+    staleTime: Infinity,
+    retry: false,
+    queryFn: ({ pageParam: pageIndex }) =>
       fetchIcebergV6<T>({
         route,
         pageSize,
-        page: pageParam,
+        page: pageIndex,
         sortBy: sorting?.id || null,
         sortReverse: sorting?.desc,
       }),
-    staleTime: Infinity,
-    retry: false,
-    getNextPageParam: () => queryParams.pageIndex,
+    getNextPageParam: (lastPage, _allPages, lastPageIndex) => {
+      if (lastPage.totalCount / pageSize > lastPageIndex) return null;
+      return lastPageIndex + 1;
+    },
+    select: (data) => {
+      const pageIndex = data.pageParams?.[data.pageParams.length - 1] ?? 0;
+      const totalCount = data?.pages[0].totalCount ?? 0;
+      return {
+        data,
```

Comment:
**Question**: what's the point to return raw `data` as the aim of a selector is to compute `data` for the view?

Reply:
It's to avoid break changes. I don't want create a bugs on a previous or actual developpement. 

If we can be sure at 100% it's ok. We can remove it. But I am not.

Maybe create a type for deprecated data on a first time and remove later ?

Reply:
I see the point 👍🏻
But I think it's kind of anti-pattern with the selector's definition so maybe we can consider to remove it in a next feature, after being sure that there is no breaking change :)